src/main/java/com/teamacronymcoders/epos/client/gui/QuiverGui.java

```diff
@@ -0,0 +1,131 @@
+package com.teamacronymcoders.epos.client.gui;
+
+import com.hrznstudio.titanium.api.client.AssetTypes;
+import com.hrznstudio.titanium.api.client.IAsset;
+import com.hrznstudio.titanium.api.client.IGuiAddon;
+import com.hrznstudio.titanium.client.gui.IGuiAddonConsumer;
+import com.hrznstudio.titanium.client.gui.addon.ICanMouseDrag;
+import com.hrznstudio.titanium.client.gui.addon.IClickable;
+import com.hrznstudio.titanium.client.gui.addon.SlotsGuiAddon;
+import com.hrznstudio.titanium.client.gui.asset.IAssetProvider;
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.teamacronymcoders.epos.container.QuiverContainer;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.IHasContainer;
+import net.minecraft.client.gui.screen.inventory.ContainerScreen;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.inventory.container.INamedContainerProvider;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import org.lwjgl.glfw.GLFW;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QuiverGui extends ContainerScreen<QuiverContainer> implements IGuiAddonConsumer, IHasContainer<QuiverContainer>, INamedContainerProvider {
+    private final QuiverContainer container;
+    private IAssetProvider assetProvider;
+    private int x;
+    private int y;
+    private List<IGuiAddon> addons;
+
+    private boolean isMouseDragging;
+    private int dragX;
+    private int dragY;
+
+    public QuiverGui(QuiverContainer container, PlayerInventory inventory, ITextComponent title) {
+        super(container, inventory, new TranslationTextComponent("gui.epos.quiver"));
+        this.container = container;
+        this.assetProvider = IAssetProvider.DEFAULT_PROVIDER;
+        IAsset background = IAssetProvider.getAsset(assetProvider, AssetTypes.BACKGROUND);
+        this.xSize = background.getArea().width;
+        this.ySize = background.getArea().height;
+        this.addons = new ArrayList<>();
+        addons.add(new SlotsGuiAddon(container.getHandler()));
+    }
+
+    @Override
+    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
```

Comment:
Does Titanium really not handle a lot of this stuff?

Reply:
Nope, this is why the GUI was such a bitch.
Currently Titanium only has a TileEntity specific implementation.
I had to do this implementation to get it working on an item :P

Reply:
So what I'm hearing is PR time?